source/common/router/router.h

```diff
@@ -309,7 +309,16 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   // Called immediately after a non-5xx header is received from upstream, performs stats accounting
   // and handle difference between gRPC and non-gRPC requests.
   void handleNon5xxResponseHeaders(const Http::HeaderMap& headers, bool end_stream);
-  void sendLocalReply(Http::Code code, const std::string& body, bool overloaded);
+
+  /**
+   * Send a locally generated (non-proxied) HTTP response.
+   * @param code supplies the HTTP status code.
+   * @param body supplies the response body (empty string if no body is needed).
+   * @param add_headers supplies an optional callback function that can add additional
+   *                    headers to the response.
```

Comment:
that can alter the headers in the response
since it could also theoretically remove or just change, yeah?